dts/arm/atmel/same70.dtsi

```diff
@@ -352,11 +352,18 @@
 			pinctrl-0 = <&pd0a_gmac_gtxck &pd1a_gmac_gtxen
 				     &pd2a_gmac_gtx0 &pd3a_gmac_gtx1
 				     &pd4a_gmac_grxdv &pd5a_gmac_grx0
-				     &pd6a_gmac_grx1 &pd7a_gmac_grxer
-				     &pd8a_gmac_gmdc &pd9a_gmac_gmdio>;
+				     &pd6a_gmac_grx1 &pd7a_gmac_grxer>;
 		};
 
-		tc0: tc@4000c000 {
+		mdio: mdio@40050088 {
```

Comment:
The MDIO driver gets the register address from the device tree and it does share the same register space as the Ethernet node. I don't think there is a way to avoid that warning, other than hardcoding the register address into the MDIO driver and putting a throwaway address in the device tree for the MDIO node.

Reply:
>  I don't think there is a way to avoid that warning, other than hardcoding the register address into the MDIO driver and putting a throwaway address in the device tree for the MDIO node.

You are free to adjust the dtc warning flags; see e..g https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/nrf5340dk_nrf5340/pre_dt_board.cmake.

Reply:
>  if there is no necessity to make a similar tree as below, great!

@nandojve You can definitely make `/soc/foo@deadbeef` and `/soc/bar@deadbeef` nodes with the same unit addresses and skip the warning using pre_dt_board.cmake like I linked to above. Nordic SoCs have the same issue where we have multiple nodes with the same unit address and that's how we silence it.

Reply:
@nandojve - I've kep the node address as `0x40050088` to avoid conflict but kept the `reg` value as `0x40050000` . The `reg` value is picked up by the driver.

Reply:
> I've kep the node address as `0x40050088` to avoid conflict but kept the `reg` value as `0x40050000`

Sorry @arvinf, but that's forbidden by the devicetree specification:

> The unit-address must match the first address specified in the reg property of the node.

Reply:
I plan on changing this per suggestion from @nandojve and convert it to a nested structure where the MDIO node is inside the Ethernet node and doesn't need it's own register adderss.